src/main/java/io/confluent/connect/elasticsearch/ElasticsearchClient.java

```diff
@@ -484,15 +488,18 @@ private boolean createIndex(String index) {
    *
    * @param response    the response to process
    * @param executionId the execution id of the request
+   * @return true if the record was not successfully processed, and we should not commit its offset
    */
-  private void handleResponse(BulkItemResponse response,
+  private boolean handleResponse(BulkItemResponse response,
                               long executionId) {
     if (response.isFailed()) {
       for (String error : MALFORMED_DOC_ERRORS) {
         if (response.getFailureMessage().contains(error)) {
-          handleMalformedDocResponse(response);
-          reportBadRecord(response, executionId);
-          return;
+          boolean failed = handleMalformedDocResponse(response);
+          if (!failed) {
+            reportBadRecord(response, executionId);
```

Comment:
Note for reviewer: I'm also fixing DLQ reporting in this scenario: only send to DLQ if we are not failing the task (and eventually retry the record).